state/protocol/badger/mutator_test.go

```diff
@@ -131,7 +131,7 @@ func TestSealedIndex(t *testing.T) {
 		require.NoError(t, err)
 
 		// build a chain:
-		// G <- B1 <- B2 (resultB1) <- B3 <- B4 (resultB2, resultB3) <- B5 (sealB1) <- B6 (sealB2, sealB3) <- B7
+		// G <- B1 <- B2 (resultB1(vb1)) <- B3 <- B4 (resultB2(vb2), resultB3(vb3)) <- B5 (sealB1) <- B6 (sealB2, sealB3) <- B7
```

Comment:
Its a matter of opinion of course, but I don't find this overly complicated. Test itself is rather easy, its just that we need to build a long chain of blocks to test all the aspects - so its input or test data issue.
I don't think separate test would be any simpler, as the gist of test is to check if its properly mutates the state, so if not by input the same date would have to be mocked or initial state somehow prepared.

Reply:
I agree that adding a separate test case would need to generate similar state to test against, and would almost certainly be more code in total. My point is about keeping the scope of tests limited, so each one is more understandable and isolated on its own, even if we have more test code in aggregate.

`TestSealedIndex` is responsible for testing that the index of sealed blocks is updated correctly as we incorporate new blocks into the state. Many other things can happen when we seal a block, besides updating this index, one of which now is updating the `VersionTable`.

However, we don't include assertions for everything that might happen when sealing a block into `TestSealedIndex`, because that would make the test very complicated. The test would be more difficult to understand, because it is longer and dealing with more concepts. If the test fails, it is more difficult to determine what caused the failure - the sealing index? or something obliquely related to sealing which was bolted on to the test for convenience?

After the changes in this PR, `TestSealedIndex` is 146 lines long and mentions by my count 16 named entities in its test data. It's a complicated test - many tests in this package are. Some of the complexity is necessary, but some of it can be mitigated by isolating distinct functionality into distinct test cases.